recipes/doxygen/all/conanfile.py

```diff
@@ -88,33 +75,31 @@ def package_id(self):
         self.compatible_packages.append(compatible_pkg)
 
     def source(self):
-        tools.files.get(self, **self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["build_parse"] = self.options.enable_parse
-        self._cmake.definitions["build_search"] = self.options.enable_search
-        self._cmake.definitions["use_libc++"] = self.settings.compiler.get_safe("libcxx") == "libc++"
-        self._cmake.definitions["win_static"] = "MT" in self.settings.compiler.get_safe("runtime", "")
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def build(self):
         if os.path.isfile("Findflex.cmake"):
             os.unlink("Findflex.cmake")
         if os.path.isfile("Findbison.cmake"):
             os.unlink("Findbison.cmake")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.files.patch(self, **patch)
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        if Version(self.version) < "1.9.0":
+            replace_in_file(self, "addon/doxysearch/CMakeLists.txt",
+                            "find_package(Xapian REQUIRED)", "find_package(xapian REQUIRED)")
+            replace_in_file(self, "addon/doxysearch/CMakeLists.txt",
+                            "\"uuid.lib rpcrt4.lib ws2_32.lib\"", "uuid.lib rpcrt4.lib ws2_32.lib")
+            rm(self, "FindXapian.cmake", "cmake")
+        cmake = CMake(self)
+        cmake.definitions["build_parse"] = self.options.enable_parse
+        cmake.definitions["build_search"] = self.options.enable_search
+        cmake.definitions["use_libc++"] = self.settings.compiler.get_safe("libcxx") == "libc++"
+        cmake.definitions["win_static"] = "MT" in self.settings.compiler.get_safe("runtime", "")
```

Comment:
There's helper for is static runtime you can use here

Also this old helper benefits from the cache that was previously done, not sure why it's removed. Was the a reason I might have missed?